azkaban-executor/azkaban-emr-jobtype/src/main/java/uk/gov/dwp/dataworks/azkaban/services/CompositeService.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.dwp.dataworks.azkaban.services;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.elasticmapreduce.AmazonElasticMapReduce;
+import uk.gov.dwp.dataworks.azkaban.model.InvocationPayload;
+import uk.gov.dwp.dataworks.azkaban.model.InvocationResult;
+import uk.gov.dwp.dataworks.azkaban.utility.EmrUtility;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class CompositeService extends DelegateService {
```

Comment:
Why is this called CompositeService? It isnt generic enough to be named that. Similarly DelegateService is just a wrapper to add info messages and doesnt Delegate.

Reply:
It's CompositeService because it is composed of other services. It's delegate service because it is delegated to, it is a delegate, it doesn't delegate itself.